force-app/main/default/lwc/rd2ChangeHistory/rd2ChangeHistory.js

```diff
@@ -0,0 +1,87 @@
+import { LightningElement, api, track, wire } from 'lwc';
+import getChangeHistory from '@salesforce/apex/RD2_ChangeHistoryController.getChangeHistory'
+import { NavigationMixin } from 'lightning/navigation';
+
+import commonViewMore from '@salesforce/label/c.commonViewMore';
+import commonError from '@salesforce/label/c.commonError';
+import rdchDisabled from '@salesforce/label/c.RDCH_Disabled';
+import rdchChangeHistory from '@salesforce/label/c.RDCH_Change_History';
+import rdchNoRecords from '@salesforce/label/c.RDCH_No_Records';
+
+const STATES = {
+    LOADING: 'LOADING',
+    READY: 'READY',
+    ERROR: 'ERROR',
+    SETTING_DISABLED: 'SETTING_DISABLED',
+    EMPTY_LIST: 'EMPTY_LIST'
+};
+
+export default class Rd2ChangeHistory extends NavigationMixin(LightningElement) {
+    @api recordId;
+    @api changeTypeFilter;
+    @api recordLimit;
+
+    @track changeHistory;
+
+    displayState = STATES.LOADING;
+    errorMessage;
+
+    labels = {
+        commonError,
+        commonViewMore,
+        rdchDisabled,
+        rdchChangeHistory,
+        rdchNoRecords
+    };
+
+    async connectedCallback() {
+        const { recordId, changeTypeFilter, recordLimit } = this;
+        try {
+            this.changeHistory = await getChangeHistory({ recurringDonationId: recordId, changeTypeFilter, recordLimit });
```

Comment:
I think the order is wrong here? `changeTypeFilter` should be after `recordLimit`

Reply:
Arguments for imperative Apex calls have to be an object, and the shorthand makes it look like it's a list of params here. I'll re-format this for clarity.

Reply:
Hmm, well I couldn't get Change Type filtering to work and thought maybe this was the culprit.

Reply:
Hey I was right, updating to `await getChangeHistory({ recurringDonationId: recordId, recordLimit: recordLimit, filter: changeTypeFilter})` fixes is (you're right though, order doesn't matter)

Reply:
Aha, thanks for spotting that! I changed the Controller args to match what's in the LWC, and reformatted that call so it is a bit easier to read.